static/app/views/performance/cache/tables/transactionsTable.tsx

```diff
@@ -0,0 +1,165 @@
+import {DataTitles} from 'sentry/views/starfish/views/spans/types';
+
+type Row = Pick<
+  MetricsResponse,
+  | 'project.id'
+  | 'transaction'
+  | 'spm()'
+  | 'cache_hit_rate()'
+  | 'sum(span.self_time)'
+  | 'time_spent_percentage()'
+>;
+
+type Column = GridColumnHeader<
+  'transaction' | 'spm()' | 'cache_hit_rate()' | 'time_spent_percentage()'
+>;
+
+const COLUMN_ORDER: Column[] = [
+  {
+    key: 'transaction',
+    name: t('Transaction'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: 'spm()',
+    name: `${t('Requests')} ${RATE_UNIT_TITLE[RateUnit.PER_MINUTE]}`,
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: `cache_hit_rate()`,
+    name: DataTitles.cacheMissRate,
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: 'time_spent_percentage()',
+    name: DataTitles.timeSpent,
+    width: COL_WIDTH_UNDEFINED,
+  },
+];
+
+const SORTABLE_FIELDS = ['spm()', 'cache_hit_rate()', 'time_spent_percentage()'] as const;
+
+type ValidSort = Sort & {
+  field: (typeof SORTABLE_FIELDS)[number];
+};
+
+export function isAValidSort(sort: Sort): sort is ValidSort {
+  return (SORTABLE_FIELDS as unknown as string[]).includes(sort.field);
+}
+
+interface Props {
+  data: Row[];
+  isLoading: boolean;
+  sort: ValidSort;
+  error?: Error | null;
+  meta?: EventsMetaType;
+  pageLinks?: string;
+}
+
+export function TransactionsTable({
+  data,
+  isLoading,
+  error,
+  meta,
+  pageLinks,
+  sort,
+}: Props) {
+  const location = useLocation();
+  const organization = useOrganization();
+
+  const handleCursor: CursorHandler = (newCursor, pathname, query) => {
+    browserHistory.push({
+      pathname,
+      query: {...query, [QueryParameterNames.TRANSACTIONS_CURSOR]: newCursor},
+    });
+  };
+
+  return (
+    <Fragment>
+      <GridEditable
+        aria-label={t('Transactions')}
+        isLoading={isLoading}
+        error={error}
+        data={data}
+        columnOrder={COLUMN_ORDER}
+        columnSortBy={[
+          {
+            key: sort.field,
+            order: sort.kind,
+          },
+        ]}
+        grid={{
+          renderHeadCell: col =>
+            renderHeadCell({
+              column: col,
+              sort,
+              location,
+              sortParameterName: QueryParameterNames.TRANSACTIONS_SORT,
+            }),
+          renderBodyCell: (column, row) =>
+            renderBodyCell(column, row, meta, location, organization),
+        }}
+        location={location}
+      />
+
+      <Pagination pageLinks={pageLinks} onCursor={handleCursor} />
+    </Fragment>
+  );
+}
+
+function renderBodyCell(
+  column: Column,
+  row: Row,
+  meta: EventsMetaType | undefined,
+  location: Location,
+  organization: Organization
+) {
+  if (column.key === 'transaction') {
+    return (
+      <TransactionCell
+        project={String(row['project.id'])}
+        transaction={row.transaction}
+        transactionMethod={row['transaction.method']}
+      />
+    );
```

Comment:
😭 so much custom `transaction` handling in every module! Not your fault, I might standardize that soon